pkg/api/product_cache.go

```diff
@@ -0,0 +1,91 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
+)
+
+// ProductAccess models the response from the Fastly Product Entitlement API
+type ProductAccess struct {
```

Comment:
It looks like `Product` is only used to be embedded in `ProductAccess`. This is a nit, but I would merge `ProductAccess` and `Product` to just `Product`.